base/logger_external.go

```diff
@@ -0,0 +1,180 @@
+package base
+
+import (
+	"github.com/couchbase/gocb"
+	"github.com/couchbase/goutils/logging"
+	"gopkg.in/couchbase/gocbcore.v7"
+)
+
+// This file implements wrappers around the loggers of external packages
+// so that all of SG's logging output is consistent
+
+func initExternalLoggers() {
+	gocb.SetLogger(GoCBLogger{})
+	gocbcore.SetLogger(GoCBCoreLogger{})
+	logging.SetLogger(CBGoUtilsLogger{})
+}
+
+// **************************************************
+// Implementation of github.com/couchbase/gocb.Logger
+// **************************************************
+type GoCBLogger struct{}
+
+var _ gocb.Logger = GoCBLogger{}
+
+func (GoCBLogger) Log(level gocb.LogLevel, offset int, format string, v ...interface{}) error {
+	switch level {
+	case gocb.LogError:
+		Errorf(KeyGoCB, format, v...)
+	case gocb.LogWarn:
+		Warnf(KeyGoCB, format, v...)
+	default:
+		Tracef(KeyGoCB, format, v...)
+	}
+	return nil
+}
+
+// ******************************************************
+// Implementation of github.com/couchbase/gocbcore.Logger
+// ******************************************************
+type GoCBCoreLogger struct{}
+
+var _ gocbcore.Logger = GoCBCoreLogger{}
+
+func (GoCBCoreLogger) Log(level gocbcore.LogLevel, offset int, format string, v ...interface{}) error {
+	switch level {
+	case gocbcore.LogError:
+		Errorf(KeyGoCB, format, v...)
+	case gocbcore.LogWarn:
+		Warnf(KeyGoCB, format, v...)
+	default:
+		Tracef(KeyGoCB, format, v...)
+	}
+	return nil
+}
+
+// **************************************************************
+// Implementation for github.com/couchbase/goutils/logging.Logger
```

Comment:
Add a comment that we're explicitly not implementing the Map or Pair functions from the interface, because our dependency that uses this library (gomemcached) doesn't use them either.   If that changes in future we'd need to revisit.

Reply:
Done